src/parser.cpp

```diff
@@ -1174,15 +1229,16 @@ void Parser::ParserImpl::loadReset(const ResetPtr &reset, const ComponentPtr &co
                 err->setDescription("Reset referencing variable '" + variableReference +
                                     "' is not a valid reference for a variable in component '" + component->getName() + "'.");
                 err->setReset(reset);
-                err->setKind(Error::Kind::RESET);
-                err->setRule(SpecificationRule::RESET_INVALID_VARIABLE_REFERENCE);
+                err->setRule(SpecificationRule::RESET_VARIABLE_REFERENCE);
                 mParser->addError(err);
             } else {
                 reset->setVariable(referencedVariable);
             }
         } else if (attribute->isType("order")) {
-            if (convertToInt(attribute->getValue(), &order)) {
-                orderDefined = true;
+            orderDefined = true;
+            orderValid = isCellMLInteger(attribute->getValue());
+            if (orderValid) {
+                order = convertToInt(attribute->getValue());
```

Comment:
At some point, it might be good to rework this a bit, so that we can a call that looks like:

```
orderValid = convertToInt(attribute->getValue(), order);
...
```

i.e. `convertToInt()` would have the following signature: `bool convertToInt(const std::string &valueString, int &value);`